htdocs/product/list.php

```diff
@@ -467,10 +480,10 @@
 	}
 
 	//Show/hide child products. Hidden by default
-	if (!empty($conf->variants->enabled) && $search_type === 0) {
+	if (!empty($conf->variants->enabled) && !empty($conf->global->PRODUIT_ATTRIBUTES_HIDECHILD )) {
 		$moreforfilter.='<div class="divsearchfield">';
-		$moreforfilter.= '<input type="checkbox" id="search_hidechildproducts" name="search_hidechildproducts" value="on"'.($search_hidechildproducts ? 'checked="checked"' : '').'>';
-		$moreforfilter.= ' <label for="search_hidechildproducts">'.$langs->trans('HideChildProducts').'</label>';
+		$moreforfilter.= '<input type="checkbox" id="show_childproducts" name="show_childproducts"'.($show_childproducts ? 'checked="checked"':'').'>';
```

Comment:
the name of fields used into the search criteria must always start with "search_" (if not, it breaks the feature to save criteria when leaving a page and going back on it). Can you rename it into search_showchildproducts ?